jetty-server/src/main/java/org/eclipse/jetty/server/Dispatcher.java

```diff
@@ -195,8 +196,24 @@ protected void forward(ServletRequest request, ServletResponse response, Dispatc
                 baseRequest.setContextPath(_contextHandler.getContextPath());
                 baseRequest.setServletPath(null);
                 baseRequest.setPathInfo(_pathInContext);
-                if (_uri.getQuery()!=null || old_uri.getQuery()!=null)
-                    baseRequest.mergeQueryParameters(old_uri.getQuery(),_uri.getQuery(), true);
+    
+                if (_uri.getQuery() != null || old_uri.getQuery() != null)
+                {
+                    try
+                    {
+                        baseRequest.mergeQueryParameters(old_uri.getQuery(), _uri.getQuery(), true);
+                    }
+                    catch (BadMessageException e)
+                    {
+                        // Only throw BME if not in Error Dispatch Mode
+                        // This allows application ErrorPageErrorHandler to handle BME messages
+                        Boolean inErrorDispatch = (Boolean) request.getAttribute(__ERROR_DISPATCH);
+                        if(inErrorDispatch == null || !inErrorDispatch)
+                        {
+                            throw e;
+                        }
```

Comment:
I wonder if we need an else branch here that sets the query parameter to some obvious error condition?:
```
baseRequest.mergeQueryParameters("", e+"&"+e.getCause(), true);
```

Reply:
I would expect that to be a vector for XSS or something equally nefarious.

My thought process is that if the original URI is bad, then it's bad, no point attempting to fix it.
The details of what is bad is still present in the RequestDispatcher.ERROR_* attributes anyway.

But perhaps we should `log.warn()` instead if we are in this catch block?